roman_photoz/roman_catalog_handler.py

```diff
@@ -45,52 +50,71 @@ def format_catalog(self):
         Format the catalog by appending necessary fields and columns.
         """
         logger.info("Formatting catalog...")
-        # self.format_colnames()
 
-        self.catalog = self.cat_array[["id"]].copy()
-
-        for filter_name in self.filter_names:
-            self.catalog = np.lib.recfunctions.append_fields(
-                self.catalog,
-                f"flux_psf_{filter_name}",
-                self.cat_array[f"{filter_name.replace('roman_', '')}_flux_psf"],
-            )
-            self.catalog = np.lib.recfunctions.append_fields(
-                self.catalog,
-                f"flux_psf_err_{filter_name}",
-                self.cat_array[f"{filter_name.replace('roman_', '')}_flux_psf_err"],
-            )
+        self.catalog = np.empty(0, dtype=[])
+        self.catalog = rfn.append_fields(self.catalog, "label", self.cat_array["label"])
+
+        for filter_id in self.filter_names:
+            # Roman filter ID in format "fNNN"
+            flux_col_name = f"psf_{filter_id}_flux"
+            if flux_col_name in self.cat_array.dtype.names:
+                self.catalog = rfn.append_fields(
+                    self.catalog,
+                    f"psf_{filter_id}_flux",
+                    np.array(self.cat_array[flux_col_name]),
+                )
+                self.catalog = rfn.append_fields(
+                    self.catalog,
+                    f"psf_{filter_id}_flux_err",
+                    np.array(self.cat_array[f"{flux_col_name}_err"]),
+                )
 
         # populate additional required columns
-        self.catalog = np.lib.recfunctions.append_fields(
+        self.catalog = rfn.append_fields(
             self.catalog,
             "context",
-            self.cat_array["context"],
+            (
+                self.cat_array["context"]
+                if "context" in self.cat_array.dtype.names
+                else np.zeros(len(self.cat_array), dtype=int)
+            ),
             usemask=False,
         )
-        self.catalog = np.lib.recfunctions.append_fields(
+        self.catalog = rfn.append_fields(
             self.catalog,
             "zspec",
-            self.cat_array["zspec"],
+            (
+                self.cat_array["zspec"]
+                if "zspec" in self.cat_array.dtype.names
+                else np.zeros(len(self.cat_array), dtype=int)
+            ),
             usemask=False,
         )
-        self.catalog = np.lib.recfunctions.append_fields(
+        self.catalog = rfn.append_fields(
             self.catalog,
             "string_data",
-            self.cat_array["string_data"].astype(str),
+            (
+                self.cat_array["string_data"]
+                if "string_data" in self.cat_array.dtype.names
+                else np.full(len(self.cat_array), "")
+            ),
             usemask=False,
         )
+
```

Comment:
Most multiband catalogs will only have values for a few of the filters and won't have all 7.  What happens in this function in this case?  I think the right behavior is to have the appropriate columns get filled out and those columns set to 0 or -999 or something, and then for the context value to mark which fluxes contain real values and which do not.  Does that happen elsewhere?